python/genhisto.py

```diff
@@ -0,0 +1,87 @@
+import argparse
+import subprocess
+import numpy as np
+import tempfile
+import sys
+
+def main():
+
+    parser = argparse.ArgumentParser(description="Generate histogram of nifti-1 image")
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument("-nb", "--num_bins", help="The number of bins in your histogram (integer value). ", type=int)
+    group.add_argument("-br", "--bin_range", help="The interval range of the bins (float)", type=float)
+    parser.add_argument("--min_val", help="The minimum voxel value of the image", type=float)
+    parser.add_argument("--max_val", help="The maximum voxel value of the image", type=float)
+    
+
+    parser.add_argument("--num_reducers", help="The number of reducers to be used", type=int)
+    parser.add_argument("--num_mappers", help="The number of mappers to be used", type=int)
+
+    parser.add_argument("-o", "--output", help="The output folder", type=str, default="HistogramOutput")
+    
+    required = parser.add_argument_group('required arguments')
+    required.add_argument("-i", "--input_file", help="The input file", type=str, required=True)
+    required.add_argument("-hj", "--hadoop_streaming_jar", help="The hadoop streaming jar", required=True)
+
+    args = parser.parse_args()
+    
+    number_bins = args.num_bins
+    bin_range = args.bin_range
+
+    data_min = args.min_val
+    data_max = args.max_val
+
+    num_reducers = args.num_reducers
+    num_mappers = args.num_mappers
+
+    input_file = args.input_file
+    output_folder = args.output
+
+    hadoop_jar = args.hadoop_streaming_jar
+
+    
+    # will get the min/max value of the image if min/max is not already provided
+    if(data_min == None or data_max == None):
+
+        temp_dir = tempfile.mkdtemp() if 'file:///' not in output_folder else 'file:///$PWD' + tempfile.mkdtemp()
```

Comment:
i'm going to use os.getcwd() because if not i get a 'output directory already exists' error when it didn't exist prior to running my code. I feel like that might be a permissions issue, though I could be wrong